plugins/cq_cache/cq_cache.py

```diff
@@ -0,0 +1,138 @@
+import cadquery as cq
+import cadquery 
+from cadquery import exporters, importers
+from functools import wraps
+import tempfile
+import os
+import inspect
+
+from OCP.BRepTools import BRepTools
+from OCP.BRep import BRep_Builder
+from OCP.TopoDS import TopoDS_Shape
+
+
+
+
+
+TEMPDIR_PATH = tempfile.gettempdir()
+CACHE_DIR_NAME = "cadquery_geom_cache"
+CACHE_DIR_PATH = os.path.join(TEMPDIR_PATH,CACHE_DIR_NAME)
+CQ_TYPES = [cq.Shape, cq.Solid, cq.Shell, cq.Compound, cq.Face, cq.Wire, cq.Edge, cq.Vertex, TopoDS_Shape, cq.Workplane]
+
+if CACHE_DIR_NAME not in os.listdir(TEMPDIR_PATH):
+    os.mkdir(CACHE_DIR_PATH)
+
+def importBrep(file_path):
+    builder = BRep_Builder()
+    shape = TopoDS_Shape()
+    return_code = BRepTools.Read_s(shape, file_path, builder)
+    if return_code is False:
+        raise ValueError("Import failed, check file name")
+    return shape
+
+
+def get_cache_dir_size(cache_dir_path):
+    total_size = 0
+    for dirpath, dirnames, filenames in os.walk(cache_dir_path):
+        for f in filenames:
+            fp = os.path.join(dirpath, f)
+            total_size += os.path.getsize(fp)
+    return total_size
+
+def delete_oldest_file(cache_dir_path):
+    cwd = os.getcwd()
+    os.chdir(cache_dir_path)
+    files = sorted(os.listdir(os.getcwd()), key=os.path.getmtime)
+    oldest = files[0]
+    os.remove(os.path.join(cache_dir_path,oldest))
+    os.chdir(cwd)
+
+
+def build_file_name(fct, *args, **kwargs):
+    SPACER = "_"
+    file_name = fct.__name__
+    for arg in args:
+        file_name += SPACER + str(arg)
+    for kwarg_value in kwargs.values():
+        file_name += SPACER + str(kwarg_value)
+
+    return file_name
```

Comment:
> basically you're just using memory location
I'm not sure what you mean here. However I agree that using hash is a good idea, I will look more in details how it can be used.

Reply:
Sorry @Jojain, I read your code too quickly and I think I missed how you were using it. What types of arguments and keyword arguments do you intend to support in `build_file_name`?

I suspect you are going to say floats and bools, and explicitly not supporting complicated classes like Workplane or user defined classes. On my first look I saw the `CQ_TYPES = [cq.Shape, cq.Solid, cq.Shell, cq.Compound, cq.Face, cq.Wire, cq.Edge, cq.Vertex, TopoDS_Shape, cq.Workplane]` line at the top of your code and jumped to the conclusion that they were the types you were supporting, but I see that's not the case now.

My point is still the same though, you can't rely on `str(arg)` to uniquely identify an object. That's what `hash(arg)` is for (kind of).

I would recommend that this function firstly type check for any args/kwargs that are Workplanes and raise a `TypeError('Can not cache Workplane')`. Then hash every arg/kwargs, encode the hash with [base64.urlsafe_b64encode](https://docs.python.org/3/library/base64.html#base64.urlsafe_b64encode), and return that as the filename. 

The type check for Workplane is only required because `hash(cq.Workplane())` returns an int based off the id of the Workplane, which is useless for caching. Actually it's dangerous for file based caching, since between Python runs it's possible for a Workplane to wind up with the same hash/id (and string representation).

User defined classes without a `__hash__` method will run into the same problem as Workplane when used as args, but I'm not sure how to protect against that. Maybe just mention it in the doc string.

Ignore my question above @adam-urbanczyk, hashing Workplanes is not required.